qiskit_experiments/library/characterization/T2Hahn.py

```diff
@@ -0,0 +1,116 @@
+                    "t" of the pulse sequence in circuit no. i.
+                    Must be in an increasing order.
+                gate_time: time of running a single identity gate.
+                qubits: indices of the qubits whose
+                    T\ :sub:`2`:sup:`*`\ 's are to be measured.
+                n_echos: number of echo gates (`X` or `Y`).
+                phase_alt_echo: if True then alternate the echo between
+                    `X` and `Y`.
+        """
+        # Initialize base experiment
+        super().__init__(qubits)
+        self._verify_parameters(qubits, lengths, n_echos,gate_time)
+
+        # Set configurable options
+        self.set_experiment_options(lengths=list(lengths), n_echos=n_echos, phase_alt_echo=phase_alt_echo)
+        self.set_analysis_options(data_processor=probability(
+            outcome="0" * self.num_qubits))  # Need to rewrite after making the analysis class
+
+
+    def _verify_parameters(self, qubits, lengths, n_echos, gate_time):
+        """Verify input correctness, raise QiskitError if needed"""
+        if any(length <= 0 for length in lengths):
+            raise QiskitError(
+                f"The lengths list {lengths} should only contain " "positive elements."
+            )
+        if len(set(lengths)) != len(lengths):
+            raise QiskitError(
+                f"The lengths list {lengths} should not contain " "duplicate elements."
+            )
+        if any(lengths[idx - 1] >= lengths[idx] for idx in range(1, lengths)):
+            raise QiskitError(f"The number of identity gates {lengths} should " "be increasing.")
+
+        if any(qubit < 0 for qubit in qubits):
+            raise QiskitError(f"The index of the qubits {qubits} should " "be non-negative.")
+
+        if n_echos < 1:
+            raise QiskitError(f"The number of echoes {n_echos} should " "be at least 1.")
+
+        if gate_time <= 0:
+            raise QiskitError(f"The gate time {gate_time} should " "be positive.")
+
+    def _generate_circuits(self, num_of_gates: Union[List[int], np.array],
+                        gate_time: float,
+                        qubits: List[int],
+                        n_echos: int = 1,
+                        phase_alt_echo: bool = False):
+        if n_echos < 1:
+            raise ValueError('Must be at least one echo')
+
+        xdata = 2 * gate_time * np.array(num_of_gates) * n_echos
+        qr = QuantumRegister(max(qubits) + 1)
+        cr = ClassicalRegister(len(qubits))
+        circuits = []
+        for circ_index, circ_length in enumerate(num_of_gates):
+            circ = QuantumCircuit()
+            circ.name = 't2circuit_' + str(circ_index) + '_0'
+            for qind, qubit in enumerate(qubits):
+
+                # First Y90 and Y echo
+                circ.append(circ.ry(np.pi/4, [qr[qubit]]))  # Y90
+                circ = pad_id_gates(circ, qr, qubit, circ_length)  # ids
```

Comment:
don't need to pad with ID gate. please use `Delay` instruction.

Reply:
You can see example of using `Delay` in `T1` and `T2Ramsey`.

Reply:
Implemented delay instead of Identity gates.